src/systems/thermal/Thermal.cc

```diff
@@ -48,14 +51,60 @@ void Thermal::Configure(const Entity &_entity,
     gazebo::EntityComponentManager &_ecm,
     gazebo::EventManager & /*_eventMgr*/)
 {
-  if (!_sdf->HasElement("temperature"))
+  const std::string temperatureTag = "temperature";
+  const std::string heatSignatureTag = "heat_signature";
+  const std::string minTempTag = "min_temp";
+  const std::string maxTempTag = "max_temp";
+
+  if (_sdf->HasElement(temperatureTag) && _sdf->HasElement(heatSignatureTag))
+  {
+    ignwarn << "Both <" << temperatureTag << "> and <" << heatSignatureTag
+           << "> were specified, but the thermal system only uses one. "
+           << "<" << heatSignatureTag << "> will be used.\n";
+  }
+
+  if (_sdf->HasElement(heatSignatureTag))
+  {
+    std::string heatSignature = _sdf->Get<std::string>(heatSignatureTag);
+
+    // check if this is a fuel model (if so, use the heat signature path in
+    // fuel's local cache instead of the user-specified path)
+    ignition::fuel_tools::FuelClient fuelClient;
+    std::string fuelPath;
+    if (fuelClient.CachedModelFile(ignition::common::URI(heatSignature),
+          fuelPath))
+      _ecm.CreateComponent(_entity, components::SourceFilePath(fuelPath));
+    else
+      _ecm.CreateComponent(_entity, components::SourceFilePath(heatSignature));
+
+    // see if the user defined a custom temperature range
+    double min = 0.0;
+    double max = 100.0;
+    if (_sdf->HasElement(minTempTag))
+      min = _sdf->Get<double>(minTempTag);
+    else
+      ignwarn << "Missing <" << minTempTag << ">. Defaulting to 0 kelvin.\n";
+    if (_sdf->HasElement(maxTempTag))
+      max = _sdf->Get<double>(maxTempTag);
+    else
+      ignwarn << "Missing <" << maxTempTag << ">. Defaulting to 100 kelvin.\n";
```

Comment:
can we default min/max values to ambient temperature?
`min`: default to ambient if not specified. If user specified `max` and it's lower than ambient, set `min` to `max`.
`max`: default to ambient if not specified. If user specified `min` and it's higher than ambient, set `max` to `min`.

Reply:
defaults set in https://github.com/ignitionrobotics/ign-gazebo/pull/498/commits/c4c6cbbc4893b786b7e64e8c778d23662f37b2fa, feel free to take a look and let me know if this is a good approach. I added some tests too. The only concern I have is that if the user does not specify a max or min, then the temperature range is 0 because min and max are both the ambient temperature. Is that okay?

Reply:
yeah if min and max are not specified, defaulting to ambient temperature with 0 range is fine.